backend/src/openai.ts

```diff
@@ -321,31 +310,30 @@ async function performToolCalls(
 	currentLevel: LEVEL_NAMES,
 	qaLlmDefence?: QaLlmDefence
 ): Promise<ToolCallResponse> {
-	for (const toolCall of toolCalls) {
-		// only tool type supported by openai is function
-		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-		if (toolCall.type === 'function') {
-			const functionCallReply = await chatGptCallFunction(
-				toolCall.id,
-				toolCall.function,
-				currentLevel,
-				qaLlmDefence
-			);
-
-			// We assume only one function call in toolCalls, and so we return after getting function reply.
+	return toolCalls.reduce(
```

Comment:
I wonder if we should instead use Promise.all() on each tool_call so they run in parallel?

Reply:
Done.